src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -381,14 +308,13 @@ public static void SetHasExternalEventLoop(Thread thread)
 #endif
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static IntPtr GetIntPtrFromMethodHandle(RuntimeMethodHandle methodHandle)
+        public static RuntimeMethodHandle GetMethodHandleFromIntPtr(IntPtr ptr)
         {
-            var temp = new IntPtrAndHandle { methodHandle = methodHandle };
-            return temp.ptr;
+            var temp = new IntPtrAndHandle { ptr = ptr };
+            return temp.methodHandle;
         }
 
-
-        public static (string assemblyName, string className, string nameSpace, string shortClassName, string methodName) ParseFQN(string fqn)
+        public static (string assemblyName, string nameSpace, string shortClassName, string methodName) ParseFQN(string fqn)
```

Comment:
Is there a reason we're doing the splitting here in managed then calling back into native?

Reply:
To not duplicate the splitting logic in JS bit later. See previous feedback https://github.com/dotnet/runtime/pull/98391#discussion_r1489110785